packages/gatsby-image/src/index.js

```diff
@@ -207,6 +244,10 @@ class Image extends React.Component {
   }
 
   handleRef(ref) {
+    if (this.state.nativeLazyLoadSupported) {
+      // Bail because the browser natively supports lazy loading
+      return
+    }
```

Comment:
This ignores the logic below that updates the `imgLoaded` and `imgCached` state after the image component is mounted(and thus the ref fires to make the actual image visible(`isVisible`) and begin loading).

I understand that `isVisible` is being set to `true` earlier, but `imgLoaded` needs to be set to reveal the image when it's finished loading in with a fade-in effect(unless marked critical or fade-in disabled). `imgCached` additionally avoids the transition if there's no loading delay.

Perhaps `imgCached` needs to be added to `handleImageLoaded()`? When I implemented `imgCached` it didn't seem necessary but I hadn't tested with other configurations such as with Interaction Observer disabled. This is the only other method it seems `imgLoaded` is set.

Reply:
We're good to go because the code in `handleRef` only handles the IO logic. onLoad for the image is called in `handleImageLoaded` (which will be called irrespective)